azurerm/resource_arm_template_deployment.go

```diff
@@ -152,6 +155,22 @@ func resourceArmTemplateDeploymentCreateUpdate(d *schema.ResourceData, meta inte
 		Properties: &properties,
 	}
 
+	deploymentValidateResponse, err := client.Validate(ctx, resourceGroup, name, deployment)
+
+	if err != nil {
+		return fmt.Errorf("Error on communicating with portal to validate Template for Deployment: %+v", err)
+	}
+
+	if deploymentValidateResponse.Error != nil {
+		if deploymentValidateResponse.Error.Message != nil {
+			return fmt.Errorf("Error validating Template for Deployment: %+v", deploymentValidateResponse.Error.Message)
```

Comment:
```suggestion
			return fmt.Errorf("Error validating Template for Deployment %q (Resource Group %q): %+v", name, resourceGroup, *deploymentValidateResponse.Error.Message)
```

(we also need to dereference this)